ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -0,0 +1,376 @@
+        ])
+
+        return bottomSheetView
+    }()
+
+    // MARK: - Gesture handling
+
+    @objc private func handleResizingHandleViewTap(_ sender: UITapGestureRecognizer) {
+        if currentOffsetFromBottom != collapsedOffsetFromBottom {
+            animate(to: .collapsed, velocity: 0)
+            hostedScrollView?.setContentOffset(.zero, animated: true)
+        } else {
+            animate(to: .expanded, velocity: 0)
+        }
+    }
+
+    @objc private func handlePan(_ sender: UIPanGestureRecognizer) {
+        switch sender.state {
+        case .began:
+            stopAnimationIfNeeded()
+            fallthrough
+        case .changed:
+            translateSheet(by: sender.translation(in: view))
+            sender.setTranslation(.zero, in: view)
+        case .ended, .cancelled, .failed:
+            completePan(with: sender.velocity(in: view).y)
+        default:
+            break
+        }
+    }
+
+    private func translateSheet(by translationDelta: CGPoint) {
+        let maxOffset = expandedOffsetFromBottom + Constants.maxRubberBandOffset
+        let minOffset = collapsedOffsetFromBottom - Constants.maxRubberBandOffset
+
+        var offsetDelta = translationDelta.y
+        if currentOffsetFromBottom <= collapsedOffsetFromBottom || currentOffsetFromBottom >= expandedOffsetFromBottom {
+            offsetDelta *= translationRubberBandFactor(for: currentOffsetFromBottom)
+        }
+        bottomSheetOffsetConstraint.constant = -min(max(currentOffsetFromBottom - offsetDelta, minOffset), maxOffset)
+    }
+
+    private func translationRubberBandFactor(for currentOffset: CGFloat) -> CGFloat {
+        var offLimitsOffset: CGFloat = 0.0
+        if currentOffset > expandedOffsetFromBottom {
+            offLimitsOffset = min(currentOffset - expandedOffsetFromBottom, Constants.maxRubberBandOffset)
+        } else if currentOffset < collapsedOffsetFromBottom {
+            offLimitsOffset = min(collapsedOffsetFromBottom - currentOffset, Constants.maxRubberBandOffset)
+        }
+
+        return max(1.0 - offLimitsOffset / Constants.maxRubberBandOffset, Constants.minRubberBandScaleFactor)
+    }
+
+    // MARK: - Animations
+
+    private func completePan(with velocity: CGFloat) {
+        if abs(velocity) < Constants.directionOverrideVelocityThreshold {
+            // Velocity too low, snap to the closest offset
+            if abs(collapsedOffsetFromBottom - currentOffsetFromBottom) < abs(expandedOffsetFromBottom - currentOffsetFromBottom) {
+                move(to: .collapsed, velocity: velocity)
+            } else {
+                move(to: .expanded, velocity: velocity)
+            }
+        } else {
+            // Velocity high enough, animate to the offset we're swiping towards
+            if velocity > 0 {
+                move(to: .collapsed, velocity: velocity)
+            } else {
+                move(to: .expanded, velocity: velocity)
+            }
```

Comment:
this is heavily nested code so could flatten by slightly de-duping: 
```
let state = velocity > 0 ? .collapsed : .expanded
move(to: state, velocity: velocity)
```
Same with above